### PR TITLE
KBHDP SOA Case Study

### PR DESCRIPTION
**Pull request by zacharybinger**
Creating PR for SOA Case Study

### CODE REVIEW

File: src/watertap_contrib/reflo/analysis/case_studies/KBHDP/KBHDP_SOA.py [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+]
+
+
+def propagate_state(arc):
+    _prop_state(arc)
+    # print(f"Propogation of {arc.source.name} to {arc.destination.name} successful.")
+    # arc.source.display()
+    # print(arc.destination.name)
+    # arc.destination.display()
+    # print("\n")
```
Comment (adam-a-a) [inline src/watertap_contrib/reflo/analysis/case_studies/KBHDP/KBHDP_SOA.py:69]: delete?

File: src/watertap_contrib/reflo/analysis/case_studies/KBHDP/KBHDP_SOA.py [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+
+    m.fs.pump_to_ro = Arc(
+        source=m.fs.pump.outlet,
+        destination=m.fs.RO.feed.inlet,
+    )
+
+    # m.fs.translator_to_ro = Arc(
+    #     source=m.fs.TDS_to_NaCl_translator.outlet,
+    #     destination=m.fs.RO.feed.inlet,
+    # )
```
Comment (adam-a-a) [inline src/watertap_contrib/reflo/analysis/case_studies/KBHDP/KBHDP_SOA.py:210]: delete?

File: src/watertap_contrib/reflo/analysis/case_studies/KBHDP/components/__init__.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from .softener import *
 from .UF import *
 from .ro_system import *
+from .deep_well_injection import *
```
Comment (adam-a-a) [inline src/watertap_contrib/reflo/analysis/case_studies/KBHDP/components/__init__.py:4]: Typically I would think we'd be careful with wildcard imports.

**Review by adam-a-a — APPROVED** [review]
LGTM - just some cleanup if you want to polish before merge

File: src/watertap_contrib/reflo/analysis/case_studies/KBHDP/KBHDP_SOA.py (hunk 2 above)
Comment (zacharybinger) [reply src/watertap_contrib/reflo/analysis/case_studies/KBHDP/KBHDP_SOA.py:210]: Yup!

**Review by kurbansitterley — APPROVED** [review]
LGTM - tests and clean up/finalization once we have final results.

**Pull request merged**